### PR TITLE
Introduce codenvy/che-test and codenvy/che-file docker images

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

It provides two docker images
**codenvy/che-test**
`docker run --rm -v /var/run/docker.sock:/var/run/docker.sock codenvy/che-test <post-check> [URL of Che/Codenvy] [login] [password]
`
Which create a workspace, start it and display log

**codenvy/che-file**
still in wip with a run syntax close to
docker run -v /var/run/docker.sock:/var/run/docker.sock -v "$PWD":"$PWD" --rm codenvy/che-file $PWD <init|up>
### Tests written?

No

### CODE REVIEW

**Comment by TylerJewell** [comment]
@benoitf - I was able to build the che-test image and it worked great.  What should we do to get an initial commit merged in?  By having an initial commit, we can then initiate CI build systems to get some images hosted for furtther testing.  I'll finish, #1970 once these images are hosted.

File: dockerfiles/lib/typescript/src/typings/es6-promise/es6-promise.d.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// Type definitions for es6-promise
+// Project: https://github.com/jakearchibald/ES6-Promise
+// Definitions by: François de Campredon <https://github.com/fdecampredon/>, vvakame <https://github.com/vvakame>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
```
Comment (skabashnyuk) [inline dockerfiles/lib/typescript/src/typings/es6-promise/es6-promise.d.ts]: What is the origin of this file?

File: dockerfiles/lib/typescript/src/typings/requirejs/require.d.ts [hunk 2]
```diff
@@ -0,0 +1,397 @@
+// Type definitions for RequireJS 2.1.20
```
Comment (skabashnyuk) [inline dockerfiles/lib/typescript/src/typings/requirejs/require.d.ts]: Do we have CQ for this file?

Comment (benoitf) [inline dockerfiles/lib/typescript/src/typings/requirejs/require.d.ts]: @skabashnyuk I'm not sure we need CQ as it's to check the compilation, not the runtime
Or what I could do is to remove .ts file and let it be downloaded when compiling project

File: dockerfiles/lib/typescript/src/typings/es6-promise/es6-promise.d.ts (hunk 1 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/typings/es6-promise/es6-promise.d.ts]: deleted check-in, will be part of only a compile time type dependencies

File: dockerfiles/lib/typescript/src/typings/requirejs/require.d.ts (hunk 2 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/typings/requirejs/require.d.ts]: deleted check-in, will be part of only a compile time type dependencies

File: dockerfiles/che-file/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    openssl \
+    && curl -sL "https://${DOCKER_BUCKET}/builds/Linux/x86_64/docker-$DOCKER_VERSION" \
+    > /usr/bin/docker; chmod +x /usr/bin/docker
+
+COPY lib/typescript/lib /lib
+COPY lib/typescript/dependencies/runtime/node_modules /lib/node_modules
+COPY lib/typescript/bin /bin
+ADD  che-file/src/che.properties /lib/che.properties
+
+ENTRYPOINT ["/bin/che", "che-file"]
```
Comment (TylerJewell) [inline dockerfiles/che-file/Dockerfile]: New line

File: dockerfiles/che-test/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    openssl \
+    && curl -sL "https://${DOCKER_BUCKET}/builds/Linux/x86_64/docker-$DOCKER_VERSION" \
+    > /usr/bin/docker; chmod +x /usr/bin/docker
+
+
+COPY lib/typescript/lib /lib
+COPY lib/typescript/dependencies/runtime/node_modules /lib/node_modules
+COPY lib/typescript/bin /bin
+
+ENTRYPOINT ["/bin/che", "che-test"]
```
Comment (TylerJewell) [inline dockerfiles/che-test/Dockerfile]: new line

File: dockerfiles/lib/typescript/src/auth-data.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        let authData: AuthData = new AuthData(urlObject.hostname, port);
+        if (isSecured) {
+            authData.secured = true;
+        }
+        return authData;
+    }
+
+
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/auth-data.ts]: new line

File: dockerfiles/lib/typescript/src/che-file.ts [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+
+
+    self.times--;
+    if (self.times > 0 && !self.waitDone) {
+      setTimeout(self.waitCheBoot, 5000, self);
+    }
+
+
+  }
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/che-file.ts]: newline

File: dockerfiles/lib/typescript/src/dto/workspacedto.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        return this.content.id;
+    }
+
+    getName() : string {
+        return this.content.config.name;
+    }
+    getContent() : any {
+        return this.content;
+    }
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/dto/workspacedto.ts:21]: newline

File: dockerfiles/lib/typescript/src/messagebus-subscriber.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Interface required to be implemented to subscribe to message bus messages.
+ * @author Florent Benoit
+ */
+export interface MessageBusSubscriber {
+
+    handleMessage(message: string);
+
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/messagebus-subscriber.ts]: new line

File: dockerfiles/lib/typescript/src/messagebus.ts [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+                    subscriber.handleMessage(JSON.parse(jsonMessage.body));
+                });
+            }
+        }
+
+        // restart ping after received message
+        this.restartPing();
+    }
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/messagebus.ts]: newline

File: dockerfiles/lib/typescript/src/post-check.ts [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+                    reject(r);
+                });
+            });
+
+        return promise;
+    }
+
+
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/post-check.ts]: newline

File: dockerfiles/lib/typescript/src/recipebuilder.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+            var content = this.fs.readFileSync(dockerFilePath, 'utf8');
+            return content;
+        } catch (e) {
+            // file does not exist, return default
+            return this.DEFAULT_DOCKERFILE_CONTENT;
+        }
+
+    }
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/recipebuilder.ts]: newline

File: dockerfiles/lib/typescript/src/remoteip.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        var execSync = require('child_process').execSync;
+        this.ip = execSync('docker run --net host --rm codenvy/che-ip').toString().replace(/[\n\r]/g, '');
+    }
+
+
+   getIp() : string {
+       return this.ip;
+   }
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/remoteip.ts]: newline

File: dockerfiles/lib/typescript/src/workspace-event-subscriber.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+            console.log('Error when starting the workspace', message);
+            this.messageBus.close();
+            process.exit(1);
+        } else {
+            console.log('Event on workspace : ', message.eventType);
+        }
+
+    }
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/workspace-event-subscriber.ts]: newline

File: dockerfiles/lib/typescript/src/workspace-log-output-subscriber.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+ * @author Florent Benoit
+ */
+export class WorkspaceDisplayOutputMessageBusSubscriber implements MessageBusSubscriber {
+
+    handleMessage(message: string) {
+        // maybe parse data to add colors
+        console.log(message);
+    }
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/workspace-log-output-subscriber.ts]: newline

File: dockerfiles/lib/typescript/src/workspace.ts [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+
+            req.write('{}');
+            req.end();
+
+        });
+        return p;
+    }
+
+
+}
```
Comment (TylerJewell) [inline dockerfiles/lib/typescript/src/workspace.ts]: newline

File: dockerfiles/che-file/Dockerfile (hunk 3 above)
Comment (benoitf) [inline dockerfiles/che-file/Dockerfile]: fixed

File: dockerfiles/che-test/Dockerfile (hunk 4 above)
Comment (benoitf) [inline dockerfiles/che-test/Dockerfile]: fixed

File: dockerfiles/lib/typescript/src/auth-data.ts (hunk 5 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/auth-data.ts]: fixed

File: dockerfiles/lib/typescript/src/che-file.ts (hunk 6 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/che-file.ts]: fixed

File: dockerfiles/lib/typescript/src/dto/workspacedto.ts (hunk 7 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/dto/workspacedto.ts:21]: fixed

File: dockerfiles/lib/typescript/src/messagebus-subscriber.ts (hunk 8 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/messagebus-subscriber.ts]: fixed

File: dockerfiles/lib/typescript/src/messagebus.ts (hunk 9 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/messagebus.ts]: fixed

File: dockerfiles/lib/typescript/src/post-check.ts (hunk 10 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/post-check.ts]: fixed

File: dockerfiles/lib/typescript/src/recipebuilder.ts (hunk 11 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/recipebuilder.ts]: fixed

File: dockerfiles/lib/typescript/src/remoteip.ts (hunk 12 above)
Comment (benoitf) [inline dockerfiles/lib/typescript/src/remoteip.ts]: fixed

**Comment by TylerJewell** [comment]
lgtm - congratulations!

**Comment by benoitf** [comment]
if review is OK I could then merge this PR and we can keep adding changes on them as ppl will be able to test it

**Comment by TylerJewell** [comment]
ok to merge

**Pull request merged**